src/Selenium2Library/keywords/_element.py

```diff
@@ -602,41 +602,54 @@ def page_should_not_contain_image(self, locator, message='', loglevel='INFO'):
     def get_matching_xpath_count(self, xpath):
         """Returns number of elements matching `xpath`
 
-        One should not use the xpath= prefix for 'xpath'. XPath is assumed.
+        As of release 1.8 `Get Matching Xpath Count` allows the
+        argument `xpath` to start with the xpath= locator strategy.
 
-        Correct:
+        Example:
         | count = | Get Matching Xpath Count | //div[@id='sales-pop']
-        Incorrect:
         | count = | Get Matching Xpath Count | xpath=//div[@id='sales-pop']
 
         If you wish to assert the number of matching elements, use
         `Xpath Should Match X Times`.
         """
-        count = len(self._element_find("xpath=" + xpath, False, False))
-        return str(count)
+        if xpath.startswith('//'):
+            xpath = "xpath=" + xpath
+        
+        if xpath.startswith('xpath='):
+            count = len(self._element_find(xpath, False, False))
+            return str(count)
+        else:
+            raise ValueError("Element locator must be xpath and start with xpath= or //")
 
-    def xpath_should_match_x_times(self, xpath, expected_xpath_count, message='', loglevel='INFO'):
-        """Verifies that the page contains the given number of elements located by the given `xpath`.
+    def xpath_should_match_x_times(self, locator, expected_xpath_count, message='', loglevel='INFO'):
```

Comment:
I left the argument in `Get Matching Xpath Count` as _*xpath*_ but changed the argument in `Xpath Should Match X Times` to _*locator*_. Here I changed it so one wouldn't get an Robot Framework positional argument error. If, as in one of the acceptance tests, you wrote 

```RobotFramework
  Xpath Should Match X Times xpath=//input[@type="text"] ${1}
```

and the argument was _*xpath*_, Robot Framework would think you were placing a named/optional argument, _*xpath*_, before a positional argument of _*expected_xpath_count*_.

They don't have to be the same. For `Get Matching Xpath Count` since there is only one argument, which I thought _*xpath*_ was more descriptive, I left it. To avoid the unintentional error I described above I changed the argument to _*locator*_ under `Xpath Should Match X Times`.